### ISSUE TITLE
[1.12.2] Radar station emitting redstone signal if missile is targeted in safe zone, but out of alarm zone

### ISSUE THREAD

**Issue by KohlsJW3656** [BuiltBrokenModding/ICBM-Classic#138]
I've been testing the radar station and noticed that if I launch a missile targeted in the safe zone, the station will emit a redstone signal even if the missile is out of the alarm zone. This means the EMP tower or anti-ballistic missile fires too early, missing the incoming missile. This is how I tested it:

![2018-06-30_21 31 38](https://user-images.githubusercontent.com/40649064/42130534-b8af8148-7cad-11e8-9681-370ac21b848b.png)
I set up a firing station 300 blocks away from my target.

![2018-06-30_21 39 44](https://user-images.githubusercontent.com/40649064/42130549-23887146-7cae-11e8-8389-3074b31daf5f.png)
The target is in red, and the missile launch location is in blue.

![2018-06-30_21 31 59](https://user-images.githubusercontent.com/40649064/42130563-5966eef0-7cae-11e8-9090-a71c9333402c.png)
I aligned the path with chunk loaders.

![2018-06-30_21 34 49](https://user-images.githubusercontent.com/40649064/42130567-898f070c-7cae-11e8-9be5-43c07d8f48e0.png)
I then targeted the missile in the safe zone and fired. Notice that the redstone signal is on, yet the missile is outside the range of the radar station. (The missile is not in the alarm zone at this point. The station should not be emitting a signal)

![2018-06-30_21 35 08](https://user-images.githubusercontent.com/40649064/42130583-ea99fdea-7cae-11e8-9ae9-8f29d2cbd5d5.png)
The redstone signal would turn off and resume when the missile, the red dot, came in range of the radar station, activating the EMP tower.

![2018-06-30_22 16 25](https://user-images.githubusercontent.com/40649064/42130742-440482d4-7cb3-11e8-91d4-5b6595c7d3b3.png)
This map shows that the missile was detected by the station at the yellow point, and the EMP tower neutralized it at the pink point.

To summarize, the radar station is detecting a missile targeted in the safe zone, but it's completely disregarding if the missile is in range of the alarm zone.

**Comment by DarkGuardsman** [comment]
Alarm zone is an old feature that no longer works. I think in the original version it was meant to trigger redstone. While the safe zone triggered AB missiles only.

 Its likely to be removed or updated to work in a different way. Especially with the planned overhaul to the AB missile system.

**Comment by DarkGuardsman** [comment]
Next version should have it fixed. Alarm Zone has been changed to Detection Range. Safe Zone has been changed to trigger range. How it works now is that detection range will define how much area to scan for missiles. Then trigger range will set off redstone and AB missiles if the missile will impact in the range. To get a good effect set EMP towers and AB to slightly larger than detection range.
